File: python/chapter1/exercise2.py
def run(input_data):
    input_len = len(input_data)
    input_list = list(input_data)
    med = input_len/2
    for i in range(0, med):
        temp_val = input_list[i]
        input_list[i] = input_list[input_len-i-1]
        input_list[input_len-i-1] = temp_val
    return ''.join(input_list)

File: cpp/chapter1/Exercise2.cpp
void run(char *str)
{
    int i = -1;
    char a;
    do {
        i++;
        a = str[i];
    } while (a != '\0');

    for (int j = 0; j < i/2; j++) {
        char tempChar = str[j];
        str[j] = str[i-j-1];
        str[i-j-1] = tempChar;
    }
}
